templates/federated_reporting/mrproper.py

```diff
@@ -0,0 +1,174 @@
+#!/bin/python3
+# mrproper.py - script to remove hosts on this feeder
+# which have a newer incoming contact time on another
+# feeder which is managed by the same superhub.
+#
+# To aid in setup, first enable mrproper by setting a class in augments (def.json).
+# This enables policy in cfe_internal/enterprise/federation/federation.cf
+#
+# ```json
+# {
+#   "classes": {
+#     "cfengine_mp_fr_enable_mrproper": [ "any::" ]
+#   }
+# }
+# ```
+#
+# After that policy has run on superhub and feeders, create
+# a mrproper account on superhub and feeders and use cf-secret
+# to encrypt the credentials. The policy will handle distribution
+# of certificate files and credentials to all hubs.
+#
+# On the superhub execute the following command to encrypt credentials
+# in the /opt/cfengine/federation/cftransport/mrproper directory.
+#
+# ```sh
+# cf-secret encrypt -k $(ls *.pub | paste -sd,) -o superhub-mrproper.secret -
+# (enter credentials and type Ctrl-D to finish)
+# ```
+
+import subprocess
+import urllib3
+import json
+import platform
+
+# Primary dir in which to place various needed files
+MRPROPER_DIR = "/opt/cfengine/federation/mrproper"
+
+# collect cert files from /var/cfengine/httpd/ssl/certs on
+# superhub and feeders and cat all together into hubs.cert
+CERT_PATH = MRPROPER_DIR + "/hubs.cert"
+
+# create a mrproper role
+# on the superhub give the role Query API: Run query access
+# on the feeder give the role Hosts: Delete, Host list, and View host access
+# create a mrproper account on superhub and feeders with only mrproper role
+FEEDER_API_USER = "mrproper"
+SUPERHUB_API_USER = "mrproper"
+
+# copy all of the feeders /var/cfengine/ppkeys/localhost.pub files
+# to a central location and encrypt the superhub mrproper account credentials
+# to all feeders.
+# cf-secret encrypt -k superhub.pub,feeder1.pub,feeder2.pub -o superhub-mrproper.cfsecret -
+# you can encrypt each feeders api credentials individually or together for ease of distribution
+# cf-secret encrypt -k superhub.pub,feeder1.pub,feeder2.pub -o feeder-mrproper.cfsecret -
+SUPERHUB_SECRET_PATH = MRPROPER_DIR + "/superhub-mrproper.cfsecret"
+FEEDER_SECRET_PATH = MRPROPER_DIR + "/feeder-mrproper.cfsecret"
```

Comment:
Both of these should reflect the hostname instead of just `superhub` and `feeder`.

Reply:
fixed.